boards/Makefile.common

```diff
@@ -96,8 +96,12 @@ all:	target/$(TARGET)/release/$(PLATFORM).bin
 .PHONY: debug
 debug:	target/$(TARGET)/debug/$(PLATFORM).bin
 
+# .PHONY: target
+target:
+	@mkdir -p target
+
 .PHONY: doc
-doc:
+doc: target
```

Comment:
The right way to do directories is use order-only targets, no need for `PHONY` here:

```
target:
    @mkdir -p target

doc: | target
```